tests/functions/test_base.py

```diff
@@ -69,3 +71,18 @@ def test_json_handler_stack_info(caplog):
     base_obj = LambdaBase(conf_file=TEST_CONF)
     base_obj.logger.exception("oops", stack_info=True)
     assert '"stack_info": "Stack (most recent call last)' in caplog.text
+
+
+def test_json_handler_with_datefmt(caplog):
+    # Ensure logger's datefmt remains configurable
+    conf = copy.deepcopy(TEST_CONF)
+    conf["logging"]["formatters"]["default"][
+        "datefmt"
+    ] = "%Y-%m-%d %H:%M:%S.%f"
+
+    # Logging shouldn't raise an error
+    LambdaBase(conf_file=conf).logger.info("Works!")
+    # Logged timestamp should be expected length
+    assert len(json.loads(caplog.text)["time"]) == len(
+        "2023-04-25 15:25:51.364875"
```

Comment:
I'm having some trouble following this test.  Maybe two related issues.

- what's the "%f" mean in the configured `datefmt` above?  It's not one of the documented `strftime` directives.
- how is this the expected length?  The `.364875` here isn't milliseconds since that should have 3 digits only.

It makes me a little suspicious that the test might pass for the wrong reason.

Is it possible to freeze the time and then check that the formatted timestamp exactly matches an expected value, rather than checking that the length matches the length of some other string?

Reply:
This test is just demonstrating that datefmt is configurable. So it edits the configuration to an arbitrary format different than the defaut. The logger's timeFormat now uses datetime, not time, so the `%f` seen here is datetime's microseconds. I understand the confusion, though, as that's pretty close to the new default. I'll update it to make it more distinct and add some more comments.